client/my-sites/checkout/checkout-system-decider.js

```diff
@@ -131,27 +149,30 @@ export default function CheckoutSystemDecider( {
 		}
 
 		return (
-			<CheckoutErrorBoundary
-				errorMessage={ translate( 'Sorry, there was an error loading this page.' ) }
-				onError={ logCheckoutError }
-			>
-				<StripeHookProvider fetchStripeConfiguration={ fetchStripeConfigurationWpcom }>
-					<CompositeCheckout
-						siteSlug={ siteSlug }
-						siteId={ selectedSite?.ID }
-						product={ product }
-						purchaseId={ purchaseId }
-						couponCode={ couponCode }
-						redirectTo={ redirectTo }
-						feature={ selectedFeature }
-						plan={ plan }
-						cart={ cart }
-						isWhiteGloveOffer={ isWhiteGloveOffer }
-						isComingFromUpsell={ isComingFromUpsell }
-						isLoggedOutCart={ isLoggedOutCart }
-					/>
-				</StripeHookProvider>
-			</CheckoutErrorBoundary>
+			<>
+				<CheckoutErrorBoundary
+					errorMessage={ translate( 'Sorry, there was an error loading this page.' ) }
+					onError={ logCheckoutError }
+				>
+					<StripeHookProvider fetchStripeConfiguration={ fetchStripeConfigurationWpcom }>
+						<CompositeCheckout
+							siteSlug={ siteSlug }
+							siteId={ selectedSite?.ID }
+							product={ product }
+							purchaseId={ purchaseId }
+							couponCode={ couponCode }
+							redirectTo={ redirectTo }
+							feature={ selectedFeature }
+							plan={ plan }
+							cart={ cart }
+							isWhiteGloveOffer={ isWhiteGloveOffer }
+							isComingFromUpsell={ isComingFromUpsell }
+							isLoggedOutCart={ isLoggedOutCart }
+						/>
+					</StripeHookProvider>
+				</CheckoutErrorBoundary>
+				<div id="g-recaptcha"></div>
```

Comment:
I assume that this component is designed to be overwritten using DOM mutation directly by some other JS code. Since this is being rendered by React, that presents some risk. When re-rendering, React may try to re-render this (empty) element also, which could create unexpected effects.

It might be best to create a custom React component here for the Recaptcha that controls its own rendering so that the component can be re-rendered by its parent but will allow its internal html to be controlled by DOM manipulation. It might also be a good place to put that `useEffect` above; I'm not sure.

Reply:
In [ec62a58](https://github.com/Automattic/wp-calypso/pull/44231/commits/ec62a583fda9326d5c74520a4f98018a1b8ad964), I've moved the div to its own memoized component, so this component will not re-render. Please let me know if this is in line with your suggestion.

Reply:
Yup! I think that should work. Thank you!